### PR TITLE
Link hasGeometry to intermediate object by default, add --hasgeometry-as-wkt flag to enable the old behavior

### PR DESCRIPTION
**Pull request by patrickbr**
This PR adds an option `--add-geometry-objects` which lets `geo:hasGeometry` point to indermediate geometry objects. These then point to the WKT literal via `geo:asWKT`.

There are no tests yet.

### CODE REVIEW

File: include/osm2rdf/config/Constants.h [hunk 1]
```diff
@@ -201,6 +201,14 @@ const static inline std::string ADD_AREA_ORIENTED_BOUNDING_BOX_OPTION_LONG =
 const static inline std::string ADD_AREA_ORIENTED_BOUNDING_BOX_OPTION_HELP =
     "Add oriented-bounding-box to areas";
 
+const static inline std::string ADD_GEOMETRY_OBJECTS_INFO =
+    "Adding area envelopes";
```
Comment (lehmann-4178656ch) [inline include/osm2rdf/config/Constants.h:205]: "Adding intermediate geometry object"

File: include/osm2rdf/ttl/Constants.h [hunk 2]
```diff
@@ -39,6 +40,8 @@ const static inline std::string NAMESPACE__XML_SCHEMA = "xsd";
 
 // Generated constants (depending on output format)
 inline std::string IRI__GEOSPARQL__HAS_GEOMETRY;
+inline std::string IRI__GEOSPARQL__HAS_SERIALIZATION;
+inline std::string IRI__GEOSPARQL__AS_WKT;
```
Comment (lehmann-4178656ch) [inline include/osm2rdf/ttl/Constants.h:44]: Should be above "inline std::string IRI__GEOSPARQL__HAS_GEOMETRY;"

File: src/osm/FactHandler.cpp [hunk 3]
```diff
@@ -116,7 +127,15 @@ void osm2rdf::osm::FactHandler<W>::node(const osm2rdf::osm::Node& node) {
 
   _writer->writeTriple(subj, IRI__RDF_TYPE, IRI__OSM_NODE);
 
-  writeBoostGeometry(subj, IRI__GEOSPARQL__HAS_GEOMETRY, node.geom());
+  if (_config.addGeometryObjects) {
```
Comment (lehmann-4178656ch) [inline src/osm/FactHandler.cpp:130]: This replaces the current and wrong output mode, maybe the switch should be `useGeometryObjects` or something like that?

File: src/osm/FactHandler.cpp [hunk 4]
```diff
@@ -238,7 +266,15 @@ void osm2rdf::osm::FactHandler<W>::way(const osm2rdf::osm::Way& way) {
 
         _writer->writeTriple(subj, IRI__RDF_TYPE, IRI__OSM_NODE);
 
-        writeBoostGeometry(subj, IRI__GEOSPARQL__HAS_GEOMETRY, node.geom());
+        if (_config.addGeometryObjects) {
+          const std::string& geomObj = _writer->generateIRI(
+              NAMESPACE__OSM_GEOM, "node_" + std::to_string(node.id()));
+
+          _writer->writeTriple(subj, IRI__GEOSPARQL__HAS_GEOMETRY, geomObj);
+          // node geometry has already been written
```
Comment (lehmann-4178656ch) [inline src/osm/FactHandler.cpp:274]: Is this true for all nodes? I think ways can contain untagged nodes.

File: src/ttl/Writer.cpp [hunk 5]
```diff
@@ -65,11 +64,17 @@ osm2rdf::ttl::Writer<T>::Writer(const osm2rdf::config::Config& config,
       {osm2rdf::ttl::constants::NAMESPACE__OSM_WAY,
        "https://www.openstreetmap.org/way/"},
       {osm2rdf::ttl::constants::NAMESPACE__OSM_META,
-       "https://www.openstreetmap.org/meta/"}};
+       "https://www.openstreetmap.org/meta/"},
+      {osm2rdf::ttl::constants::NAMESPACE__OSM_GEOM,
+       "https://www.openstreetmap.org/geom/"}};
```
Comment (lehmann-4178656ch) [inline src/ttl/Writer.cpp:69]: We should think about moving these namespaces to `osm2rdf`.

**Comment by lehmann-4178656ch** [comment]
Thank you for the work Patrick, I've left some thoughts and minor remarks.

File: src/osm/FactHandler.cpp (hunk 4 above)
Comment (patrickbr) [reply src/osm/FactHandler.cpp:274]: Good catch!

**Comment by patrickbr** [comment]
I added two commits which

1) address the issues raised by @lehmann-4178656ch 
2) remove the osmmeta: namespace and use osm2rdf: instead
3) add a osm2rdfgeom: namespace for the intermediate geometry objects, convex hulls, envelopes and OBB
4) make the writing of intermediate geometry objects for hasGeometry the default
5) add a flag --hasgeometry-as-wkt to enable the old behavior of linking hasGeometry directly to the WKT serialization

**Review by lehmann-4178656ch — APPROVED** [review]

**Pull request merged**